dev-client/__tests__/integration/OfflineSnackbar.test.tsx

```diff
@@ -0,0 +1,304 @@
+    console.log('------ TEST: dismiss + shown when thunk fails offline ------');
+    useIsOfflineMock.mockReturnValue(true);
+    
+    const projectId = '1';
+    const initialState = {
+      project: {
+        projects: {
+          [projectId]: {
+            id: projectId,
+            name: '1',
+            privacy: 'PRIVATE',
+            archived: false,
+            updatedAt: '',
+            measurementUnits: 'METRIC',
+            description: 'desc',
+            memberships: {},
+            sites: {},
+          },
+        },
+      },
+    } as Partial<AppState>;
+    
+    // Button that simulates dispatching something to server
+    deleteProjectMock.mockReturnValue(Promise.reject(['rejected']));
+    const TestButton = () => {
+      const dispatch = useDispatch();
+      return (
+        <Button
+          title="dispatch"
+          testID="test-delete-project-btn"
+          onPress={() => {
+            console.log('PRESSED test button');
+            dispatch(deleteProject({id: projectId}));
+          }}
+        />
+      );
+    };
+
+    const screen = render(
+        <PaperProvider>
+        <Portal.Host>
+          <TestButton />
+          <OfflineSnackbar />
+          <ProjectViewScreen projectId={projectId} />
+        </Portal.Host>
+      </PaperProvider>,
+      {route: 'PROJECT_VIEW', initialState},
+    );
+    console.log("2")
+
+    const snackbar = screen.queryByTestId(snackbarTestId);
+    expect(snackbar).toBeOnTheScreen();
+ 
+    // Dismiss snackbar
+    fireEvent(snackbar, 'onDismiss');
+
+    // Question 0: If we await act, the entire suite fails in the same way as in Question 1 below. 
+    // Why would that make a difference?
+    act(() => {
+      jest.runAllTimers()
+    })
+    console.log("3")
+
+    // Question 1: Why would `expect(undefined).toBeTruthy();` here cause the entire suite to fail, not just the single test?
+    //   If put further up in the test, or in other tests, it only fails the single test, not the entire suite. 
+    //
+    // Output from `npm run test OfflineSnackbar`: 
+    //   Test suite failed to run
+    //   TypeError: Cannot set properties of undefined (setting 'message')
+    //     at node_modules/jest-circus/build/formatNodeAssertErrors.js:47:25
+    //         at Array.map (<anonymous>)
+    //   Test Suites: 1 failed, 1 total
+    //   Tests:       0 total
+    // I suspect this is the erroring line in the test library: 
+    // https://github.com/jestjs/jest/blob/main/packages/jest-circus/src/formatNodeAssertErrors.ts#L58
+    expect(screen.queryByTestId(snackbarTestId)).not.toBeOnTheScreen();
+
+
+    // Fire the test button event to make a server request. The request is mocked to fail, 
+    // which should add a message to the notificationsSlice and trigger the snackbar
+    fireEvent.press(screen.queryByTestId('test-delete-project-btn'));
+    await act(() => {
+      jest.runAllTimers();
+    });
```

Comment:
```suggestion
await act(async () => {
   fireEvent.press(screen.queryByTestId('test-delete-project-btn'));
});
```

After that, you don’t need jest.runAllTimers() to assert whether the snackbar is on the screen, since it is first mounted (i.e., it appears in the DOM) before the fade-in animation starts. This is different from the fade-out animation, where the component first fades out and only after some time returns null.